dask/_task_spec.py

```diff
@@ -656,14 +718,10 @@ def lazy_unpack(self):
 
     def inline(self, dsk) -> Task:
         self.unpack()
-        new_args = self.args.inline(dsk)
-        new_kwargs = self.kwargs.inline(dsk)
-        if isinstance(new_kwargs, DataNode):
-            kwargs = new_kwargs.value
-        else:
-            kwargs = new_kwargs.tasks
+        new_args = _inline_recursively(self.args, dsk)
+        new_kwargs = _inline_recursively(self.kwargs, dsk)
```

Comment:
Inline isn't used anywhere anymore. I thought about killing it off entirely but it's also an interesting feature. I'm fine with keeping it for a little while (I assume it is useful if we actually migrated low level fusion) but we should kill it if it is being unused